test/utils/SlotDerivation.t.sol

```diff
@@ -10,6 +10,13 @@ import {SlotDerivation} from "@openzeppelin/contracts/utils/SlotDerivation.sol";
 contract SlotDerivationTest is Test, SymTest {
     using SlotDerivation for bytes32;
 
+    function testDirtyAddress() public {
+        bytes32 result1 = SlotDerivation.deriveMapping(0, address(uint160(type(uint256).max)));
```

Comment:
This does not contain dirty upper bits. The casting to `uint160` cleans it.

Reply:
only when stored in memory afaik -- https://x.com/optimizoor/status/1560590413768994816?s=46. test reverts with current master behaviour.

Reply:
Ok

I changed the tests to something that is clearer IMO, and that support dirtying any type (bool, addres, anything else we want to add)